### PR TITLE
Basic tests added

### PR DESCRIPTION
**Pull request by iMaximal**
Work in progress ;-)

### CODE REVIEW

File: scripts/jest.setup.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 import { configure } from 'enzyme'
-import Adapter from 'enzyme-adapter-react-16'
+import ReactSixteenAdapter from 'enzyme-adapter-react-16'
```
Comment (asakalou-exadel) [inline scripts/jest.setup.ts]: I would name it EnzymeAdapter

File: src/__tests__/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+const props = {
+  authChecked: true,
+  isAuthenticated: true,
+  checkAuthentication: jest.fn(),
+  ...routeMock({}),
+}
+
+describe('<App />', () => {
+  describe('how to wrap AppContainer', () => {
+    it('should works', () => {
```
Comment (asakalou-exadel) [inline src/__tests__/App.test.tsx]: works -> work

File: src/__tests__/App.test.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    const wrapper = shallow(<App {...props} />)
+    expect(wrapper.find('ThemeProvider')).toHaveLength(1)
+    expect(wrapper.find('Suspense')).toHaveLength(1)
+    expect(wrapper.find('withI18nextTranslation(Connect(Notification))')).toHaveLength(1)
+    expect(wrapper.find('Switch')).toHaveLength(1)
+    expect(wrapper.find('RouteWithLayout')).toHaveLength(6)
+    expect(wrapper.find('Redirect')).toHaveLength(3)
+  })
+
+  it('should contain specific components if the user is not logged in via shallow', () => {
```
Comment (asakalou-exadel) [inline src/__tests__/App.test.tsx]: if the user -> if user

File: src/__tests__/App.test.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    const calculatedProps = {
+      ...props,
+      isAuthenticated: false,
+    }
+
+    const wrapper = shallow(<App {...calculatedProps} />)
+    expect(wrapper.find('ThemeProvider')).toHaveLength(1)
```
Comment (asakalou-exadel) [inline src/__tests__/App.test.tsx]: prefer search by Component rather than string name of component

File: src/__tests__/App.test.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+      expect(wrapper.find(StylesProvider)).toHaveLength(1)
+      expect(wrapper.find(Notification)).toHaveLength(1)
+      expect(wrapper.find(Switch)).toHaveLength(1)
+      expect(wrapper.find(RouteWithLayout)).toHaveLength(1)
+      expect(wrapper.find(Dashboard)).toHaveLength(0)
+      expect(wrapper.find(Login)).toHaveLength(1)
+      expect(wrapper.find(Logout)).toHaveLength(0)
+    })
+
+    test('invalid path should redirect to 404', () => {
```
Comment (asakalou-exadel) [inline src/__tests__/App.test.tsx]: what is the difference between test and it ? somewhere test somewhere it

File: src/__tests__/App.test.tsx (hunk 4 above)
Comment (iMaximal) [reply src/__tests__/App.test.tsx]: I intentionally show 2 options. Thanks.

File: src/__tests__/App.test.tsx (hunk 5 above)
Comment (iMaximal) [reply src/__tests__/App.test.tsx]: Technically there is no difference. I want to say this in the video. There is one small difference that some after **it** expect the word **should**.

File: scripts/jest.setup.ts (hunk 1 above)
Comment (iMaximal) [reply scripts/jest.setup.ts]: Thanks. ReactSixteenAdapter is prefer (if esModuleInterop: false in tsconfig).

**Pull request merged**